Sources/SentryCrash/Recording/SentryCrashBinaryImageCache.c

```diff
@@ -0,0 +1,115 @@
+#include "SentryCrashBinaryImageCache.h"
+#include "SentryCrashDynamicLinker.h"
+#include <mach-o/dyld.h>
+#include <pthread.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+typedef struct SentryCrashBinaryImageNode {
+    SentryCrashBinaryImage image;
+    bool available;
+    struct SentryCrashBinaryImageNode *next;
+} SentryCrashBinaryImageNode;
+
+static SentryCrashBinaryImageNode rootNode = { 0 };
+static SentryCrashBinaryImageNode *tailNode = NULL;
+static pthread_mutex_t binaryImagesMutex = PTHREAD_MUTEX_INITIALIZER;
+
+static void
+binaryImageAdded(const struct mach_header *header, intptr_t slide)
+{
+    pthread_mutex_lock(&binaryImagesMutex);
+    if (tailNode == NULL) {
+        pthread_mutex_unlock(&binaryImagesMutex);
+        return;
```

Comment:
`m`: If I'm not mistaken this will now always exit, as the initial value of `tailNode` is `NULL`.

Reply:
tailNode is set to rootNode at `sentrycrashbic_startCache`.

This is to prevent this function works before initializing the cache